src/ui/public/capabilities/ui_capabilities.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import chrome from 'ui/chrome';
+import { deepFreeze } from '../../../core/public/utils/deep_freeze';
+
+type Capability = boolean | Record<string, boolean>;
+
+export interface UICapabilities {
+  navLinks: Record<string, boolean>;
+  [key: string]: Record<string, boolean>;
+}
+
+const injectedUiCapabilities = chrome.getInjected('uiCapabilities');
+
+const createAccessProxy: (target: Record<string, boolean>, path?: string[]) => Capability = (
+  target: Record<string, boolean>,
+  path: string[] = []
+) =>
+  new Proxy(target, {
```

Comment:
Is there a polyfill for this in IE11?

Reply:
That's a good question.  Google [released a polyfill](https://github.com/GoogleChrome/proxy-polyfill), but [babel doesn't support polyfilling `Proxy`](https://babeljs.io/docs/en/learn#proxies).

We can use that polyfill directly by importing it into this module, if we think it's important enough to keep around.